src/controllers/handlers/explorer-handler.ts

```diff
@@ -51,81 +80,91 @@ async function fetchCombinedElements(
 ): Promise<MixedWearable[]> {
   async function fetchBaseWearables() {
     const elements = await components.baseWearablesFetcher.fetchOwnedElements(address)
+    if (!elements.length) {
+      return []
+    }
+
     const urns = elements.map((e) => e.urn)
     const entities = await components.entitiesFetcher.fetchEntities(urns)
 
-    const result: MixedBaseWearable[] = []
-    for (let i = 0; i < elements.length; ++i) {
-      const wearable = elements[i]
+    return elements.reduce<MixedBaseWearable[]>((acc, wearable, i) => {
```

Comment:
quick question: do we need a reduce here? can't we just use a `map`? It might looks less complex than this reduce